src/bp/sdk/botpress.d.ts

```diff
@@ -1183,6 +1183,48 @@ declare module 'botpress/sdk' {
     computePreviewText?: (formData: object) => string
   }
 
+  export interface FormMoreInfo {
+    label: string
+    url?: string
+  }
+
+  export interface FormAdvancedSetting {
+    key: string
+    label: string
+    type: string
+    moreInfo?: FormMoreInfo
+  }
+
+  export interface FormOption {
+    value: string
+    label: string
+    related: FormField
+  }
+
+  export interface FormContextMenu {
+    type: string
+    label: string
+  }
+
+  export interface FormField {
+    type: 'checkbox' | 'group' | 'select' | 'text' | 'textarea' | 'upload' | 'url'
+    key: string
+    label: string
+    placeholder?: string
+    options?: FormOption[]
+    fields?: FormField[]
+    group?: {
+      addLabel?: string // you have to specify the add button label
+      minimum?: number // you can specify a minimum so the delete button won't show if there isn't more than the minimum
+      contextMenu?: FormContextMenu[] // you can add a contextual menu to add extra options
+    }
+  }
+
+  export interface FormDefinition {
+    advancedSettings: FormAdvancedSetting[]
+    fields: FormField[]
+  }
+
```

Comment:
I copied this whole thing from ui-shared. I also prefixed all of the types with `form` so it's more consistent. The old definition and imports will need to be changed, but for now I haven't deleted the old one so it's not breaking.

Reply:
I'm wondering if the SDK is the correct place for those typings, since nothing in the SDK uses them. I think bp/src/common/typings.d.ts would be more appropriate for this. 

Also, we definitely should avoid duplicating definitions. If we need them somewhere else, they should be moved and used in a single place.

Common is designed to be accessible by everything: modules, backend, frontend, etc. 

What do you think ?